### PR TITLE
Task#14 DevTask: Filter favorite products card

### LINKED ISSUE #63: UX: Filter product card
- [x] #64
- [x] #66
- [x] #72
- [x] #77

### LINKED ISSUE #72: Task#14 DevTask: Filter favorite products card
## OVERVIEW
- Filter procut favorite and show list
- Can remove product from favorite list
- Can add product from favorite list
- show heart icon

### PR DESCRIPTION
**Pull request by TheHienDuong**
- Closes #72 
## OVERVIEW
User can:
- Add favorite product into list
- Add multi favorite product into list
- Remove favorite prodcuts from list in Proucst list
- Can watch favorite product in list

Feat:
- Add new prop into FavoriteProduct 
- Use bolen results to toggle heart display
- Use state to store the products id values ​​to filter and iterate through the array to display products with matching ids

## TESTING
- Add and remove Favorite products in products list (done)
- Remove Favorite products in favorite list (done)
- Icon heart removed when products in favorite list removed ( Getting an error)

## DEMO 

https://github.com/Agility-Internship/Duong-The-Hien-ReactJS/assets/113186169/15204ee6-fc19-4b7b-88a1-21876fac1732

### CODE REVIEW

File: practice-two/src/App.jsx [hunk 1]
```diff
@@ -102,7 +102,11 @@ const App = () => {
           handleManufacturerFilter={handleManufacturerFilter}
           handlePriceFilter={handlePriceFilter}
         />
-        <FavoriteProductsCard products={filterProducts} />
+        <FavoriteProductsCard
+          products={filterProducts}
+          favorites={favorites}
```
Comment (phattruong-agilityio) [inline practice-two/src/App.jsx]: favorites what? meaningless

File: practice-two/src/App.jsx [hunk 2]
```diff
@@ -102,7 +102,11 @@ const App = () => {
...
+          handleFavoriteToggle={handleFavoriteToggle}
```
Comment (phattruong-agilityio) [inline practice-two/src/App.jsx]: handleFavoriteToggle ?? please check english meaning

File: practice-two/src/components/FavoriteProductsCard.jsx [hunk 3]
```diff
@@ -13,10 +13,12 @@ import ProductCard from './ProductCard';
  * When the button is clicked, a popover with a list of favorite products is displayed.
  * The popover automatically closes when clicking outside of it.
  *
+ * @param favorites - An array of favorite product IDs
  * @param products - An array of all products
+ * @param handleFavoriteToggle - Function to handle toggling a product as favorite
  * @returns {JSX.Element} The FavoriteProductsCard Component
  */
-const FavoriteProductsCard = ({ products }) => {
+const FavoriteProductsCard = ({ products, favorites, handleFavoriteToggle }) => {
```
Comment (phattruong-agilityio) [inline practice-two/src/components/FavoriteProductsCard.jsx]: where is default value

File: practice-two/src/components/FavoriteProductsCard.jsx [hunk 4]
```diff
@@ -41,14 +43,25 @@ const FavoriteProductsCard = ({ products }) => {
...
-              {products.map((product) => (
-                <div
-                  key={product.id}
-                  className="overflow-hidden flex items-center p-2 pt-4 pb-5 border  relative rounded-xl border-gray-500"
-                >
-                  <ProductCard product={product} />
-                </div>
-              ))}
+              {favorites.map((productId) => {
+                // Find the product data from allProducts array based on productId
```
Comment (phattruong-agilityio) [inline practice-two/src/components/FavoriteProductsCard.jsx]: code style

File: practice-two/src/components/FavoriteProductsCard.jsx [hunk 5]
```diff
@@ -41,14 +43,25 @@ const FavoriteProductsCard = ({ products }) => {
...
+                const productFavorite = products.find((p) => p.id === productId);
+                if (!productFavorite) return null; // Product not found
```
Comment (phattruong-agilityio) [inline practice-two/src/components/FavoriteProductsCard.jsx]: code style

File: practice-two/src/components/FavoriteProductsCard.jsx [hunk 6]
```diff
@@ -61,8 +74,11 @@ FavoriteProductsCard.propTypes = {
   products: PropTypes.arrayOf(
     PropTypes.shape({
       id: PropTypes.number.isRequired,
+      // Define other product properties based on your ProductCard component
```
Comment (phattruong-agilityio) [inline practice-two/src/components/FavoriteProductsCard.jsx]: code style

File: practice-two/src/components/ProductCard.jsx [hunk 7]
```diff
@@ -1,6 +1,5 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-
 // Components
```
Comment (phattruong-agilityio) [inline practice-two/src/components/ProductCard.jsx]: code style

File: practice-two/src/components/ProductCard.jsx [hunk 8]
```diff
@@ -10,12 +9,14 @@ import FavoriteButton from './common/Button/ButtonFavorite';
 /**
  * Product Card Component.
  * @param product - The product data object.
+ * @param isFavoriteProduct - Boolean value indicating if the product is in favorites list.
  * @returns {JSX.Element} Product Card content
  */
-function ProductCard({ product, onSelectFavoriteSelect }) {
+function ProductCard({ product, isFavoriteProduct, onSelectFavorite }) {
```
Comment (phattruong-agilityio) [inline practice-two/src/components/ProductCard.jsx]: default value

File: practice-two/src/components/layout/ListProductsCard.jsx [hunk 9]
```diff
@@ -15,12 +15,11 @@ import Typography from '../common/Typography';
  * @param handleFavoriteToggle - Function to handle the selection/unselection of a favorite product.
  * @returns {JSX.Element} The ProductList Component.
  */
-const ProductList = ({ products, handleFavoriteToggle }) => {
+const ProductList = ({ products, handleFavoriteToggle, favorites }) => {
```
Comment (phattruong-agilityio) [inline practice-two/src/components/layout/ListProductsCard.jsx]: default value please

File: practice-two/src/components/common/Button/ButtonFavorite.jsx [hunk 10]
```diff
@@ -10,23 +10,23 @@ import Button from './Button';
 // Data
 import { ICON } from '../../../constants/data';
 
-const FavoriteButton = ({ onClick }) => {
-  // Use state to store the button's state, initialize the value to false
-  const [isFavorite, setIsFavorite] = useState(false);
+const FavoriteButton = ({ onClick, isFavorite }) => {
```
Comment (phattruong-agilityio) [inline practice-two/src/components/common/Button/ButtonFavorite.jsx]: default value please

File: practice-two/src/App.jsx (hunk 2 above)
Comment (TheHienDuong) [reply practice-two/src/App.jsx]: Yes, Changed naming for this props

File: practice-two/src/App.jsx (hunk 1 above)
Comment (TheHienDuong) [reply practice-two/src/App.jsx]: Yes, Changed naming for this props

File: practice-two/src/components/FavoriteProductsCard.jsx (hunk 3 above)
Comment (TheHienDuong) [reply practice-two/src/components/FavoriteProductsCard.jsx]: Yes, updated default value and comment for this props

File: practice-two/src/components/FavoriteProductsCard.jsx (hunk 6 above)
Comment (TheHienDuong) [reply practice-two/src/components/FavoriteProductsCard.jsx]: Yes, updated

File: practice-two/src/components/FavoriteProductsCard.jsx (hunk 5 above)
Comment (TheHienDuong) [reply practice-two/src/components/FavoriteProductsCard.jsx]: updated

File: practice-two/src/components/FavoriteProductsCard.jsx (hunk 4 above)
Comment (TheHienDuong) [reply practice-two/src/components/FavoriteProductsCard.jsx]: updated

File: practice-two/src/components/ProductCard.jsx (hunk 7 above)
Comment (TheHienDuong) [reply practice-two/src/components/ProductCard.jsx]: Done

File: practice-two/src/components/ProductCard.jsx (hunk 8 above)
Comment (TheHienDuong) [reply practice-two/src/components/ProductCard.jsx]: Yes, updated default value and comment for this props

File: practice-two/src/components/layout/ListProductsCard.jsx (hunk 9 above)
Comment (TheHienDuong) [reply practice-two/src/components/layout/ListProductsCard.jsx]: Yes, updated default value and comment for this props

File: practice-two/src/components/common/Button/ButtonFavorite.jsx (hunk 10 above)
Comment (TheHienDuong) [reply practice-two/src/components/common/Button/ButtonFavorite.jsx]: Yes,Updated

**Review by phattruong-agilityio — APPROVED** [review]

**Pull request merged**